samples/sensor/adc_cmp/README.rst

```diff
@@ -0,0 +1,27 @@
+.. _adc_cmp_npcx:
```

Comment:
I'd suggest moving this sample to a directory reflecting that it is specific to the NPCX ADC CMP (`adc_cmp` makes it sound more generic than it is).